src/Sulu/Bundle/MediaBundle/Media/Storage/StorageInterface.php

```diff
@@ -22,29 +22,37 @@
      *
      * @param $tempPath
      * @param $fileName
-     * @param $version
+     * @param $preferredStorageOptions
      *
      * @return mixed
      */
-    public function save($tempPath, $fileName, $version);
+    public function save($tempPath, $fileName, $preferredStorageOptions = null);
 
     /**
-     * Give back the path to the document.
+     * Will return a file resource.
      *
-     * @param $fileName
-     * @param $version
-     * @param $storageOption
+     * @param $storageOptions
      *
-     * @return mixed
+     * @return resource
      */
-    public function load($fileName, $version, $storageOption);
+    public function load($storageOptions);
```

Comment:
$tempPath is the php tempPath e.g. `/tmp/e5f2c235` and its not good to save the file with this name e.g. in the LocalStorage.